client/my-sites/plans-v2/use-detect-window-boundary.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { MutableRefObject, useEffect, useRef, useState } from 'react';
+import { throttle } from 'lodash';
+
+/**
+ * Returns whether `elementRef` has touched/crossed the upper Window's boundary.
+ *
+ * @param {MutableRefObject} elementRef Reference to an HTMLElement
+ * @returns {boolean} Whether the element crossed the upper boundary
+ */
+const useDetectWindowBoundary = ( elementRef: MutableRefObject< HTMLDivElement | null > ) => {
+	// Indicates whether the elementRef has crossed the upper window boundary
+	const [ borderCrossed, setBorderCrossed ] = useState( false );
+
+	// Persist its initial value since it could change if the element position is changed.
+	const elementInitialOffsetTop = useRef< number | null >( null );
+
+	const addStickyClass = useRef< ( this: Window, e: Event ) => void >(
```

Comment:
For my understanding: the use of `useRef` is only necessary because we `throttle` the function, isn't it?

Reply:
Because there is no guarantee that `throttle` is going to run once. In fact, it's probably going to run many times creating a new version of the function every time it does. Which means that the function `addStickyClassFn` added with `window.addEventListener` might not be the same one when we call `window.removeEventListener`.
```
	useEffect( () => {
		const { current: addStickyClassFn } = addStickyClass;
		window.addEventListener( 'scroll', addStickyClassFn );
		return () => window.removeEventListener( 'scroll', addStickyClassFn );
	}, [ elementRef ] );
```